stjude/proteinpaint
# User defined gene set for top variably expressed genes

## Description
Addresses the second PR defined in [this sketch](https://docs.google.com/drawings/d/1zynSoWdkLLGGrZ79uoyXW6meW81Zt1p5-vJcTMZk8y4/edit). 

User is able to select the gene set, either all genes or compare against another gene set from radio buttons. 

To test: 
1. Pull `gse.2` from sjpp
2. Use [this example](http://localhost:3000/?noheader=1&mass=%7B%22dslabel%22:%22GDC%22,%22genome%22:%22hg38%22,%22nav%22:%7B%22header_mode%22:%22hidden%22%7D,%22termfilter%22:%7B%22filter0%22:%7B%22op%22:%22and%22,%22content%22:%5B%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.disease_type%22,%22value%22:%5B%22Gliomas%22%5D%7D%7D%5D%7D%7D,%22plots%22:%5B%7B%22chartType%22:%22hierCluster%22,%22dataType%22:%22geneExpression%22,%22terms%22:%5B%7B%22gene%22:%22KIF11%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22BUB1B%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22BUB1%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22CDK1%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22CDC20%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22AURKB%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22TPX2%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22CCNB2%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22CCNA2%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22TOP2A%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22AURKA%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22KIF20A%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22KIF2C%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22CDCA8%22,%22type%22:%22geneExpression%22%7D%5D,%22termgroups%22:%5B%7B%22name%22:%22Variables%22,%22lst%22:%5B%7B%22id%22:%22case.disease_type%22%7D,%7B%22id%22:%22case.primary_site%22%7D,%7B%22id%22:%22case.demographic.gender%22%7D%5D%7D%5D%7D%5D%7D). 
3. Genes -> Edit Group -> Top Variably Expressed Genes
4. If `All protein-coding genes` is selected,  the entire list below should be returned.
5. If `MSigDB gene set` and a term from the tree is selected, only a few or possibly none of the genes below will return. For example HALLMARK_ADIPOGENESIS only returns three genes. Selecting the same term from the `MSigDB (2023.2.Hs) gene set` menu returns +100. 
6. If `Custom gene set` is selected and genes are entered separated by a comma, the number of genes returned depends on the input. Try inputing genes from this list or random genes. Inputing an erroneous gene like `123` or genes not found, shows an error message box about the gene holder that disappears in 10 seconds. 

Genes for this example: 
```[
  'COL3A1',  'COL1A1',  'EGFR',    'TNC',    'CD74',    'OLIG2',
  'PDGFRA',  'FGFR3',   'SDC4',    'NTRK2',  'CDK6',    'CDKN2A',
  'SFRP4',   'ETV4',    'ID3',     'CDKN1A', 'FKBP9',   'CDKN2C',
  'CDK4',    'CSF1R',   'PRKCB',   'PTPRT',  'ALDH2',   'ERBB3',
  'CXCR4',   'ZBTB16',  'LRP1B',   'DDR2',   'FAT1',    'WWTR1',
  'GRM3',    'ETV5',    'LMO1',    'FAT3',   'ETV1',    'EPHA3',
  'MDM2',    'NDRG1',   'CREB3L1', 'BIRC6',  'CNTNAP2', 'HLF',
  'MYCN',    'MDM4',    'CCND2',   'PTPRC',  'MLLT11',  'MAFB',
  'GAS7',    'MN1',     'TFRC',    'LCP1',   'ELN',     'EZH2',
  'GPC5',    'WNK2',    'BUB1B',   'CCND1',  'RBM15',   'HLA-A',
  'FGFR2',   'CDH10',   'SOX21',   'PTPRD',  'ERBB4',   'RGS7',
  'ARHGAP5', 'SLC45A3', 'KMT2C',   'CSMD3',  'MAP3K1',  'PER1',
  'TNFAIP3', 'RET',     'BCL7A',   'KIT',    'LIFR',    'LMO2',
  'SYK',     'B2M',     'LPP',     'PREX2',  'SHTN1',   'ROBO2',
  'PTPN13',  'BLM',     'NBEA',    'DDIT3',  'AKAP9',   'JUN',
  'LRIG3',   'KDM5A',   'PTCH1',   'CTNND2', 'MYC',     'RUNX1',
  'SOX2',    'CTNNA2',  'FBLN2',   'BCR'
]
```

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
